File: frontend/src/api/resume.ts
import api from './client'

export interface ResumePolishRequest {
  section: string
  content: string
  job_description?: string
}

export interface ResumePolishResponse {
  text: string
}

export interface ResumeAnalyzeRequest {
  resume_text: string
  job_description: string
}

export interface ResumeAnalyzeResponse {
  analysis: string
}

export interface ResumeAssistantMessage {
  role: 'user' | 'assistant'
  content: string
}

export interface ResumeAssistantPayload {
  text?: string
  audio?: Blob
  session_id: string
  resume_text: string
  job_description?: string
  interview_role?: string
  history: ResumeAssistantMessage[]
  enable_tts?: boolean
}

export interface ResumeAssistantResponse {
  text: string
  user_text: string
  mode: 'resume' | 'interview' | 'interview_summary'
  interview_question_index?: number | null
  interview_total: number
  tts_audio_url?: string | null
}

export const polishResumeText = (payload: ResumePolishRequest) =>
  api.post<ResumePolishResponse>('/resume/polish', payload)

export const analyzeResumeMatch = (payload: ResumeAnalyzeRequest) =>
  api.post<ResumeAnalyzeResponse>('/resume/analyze', payload)

export const sendResumeAssistant = (payload: ResumeAssistantPayload) => {
  const form = new FormData()
  if (payload.text) form.append('text', payload.text)
  if (payload.audio) form.append('audio', payload.audio, 'recording.wav')
  form.append('session_id', payload.session_id)
  form.append('resume_text', payload.resume_text)
  if (payload.job_description) form.append('job_description', payload.job_description)
  if (payload.interview_role) form.append('interview_role', payload.interview_role)
  form.append('history', JSON.stringify(payload.history))
  form.append('enable_tts', String(payload.enable_tts ?? true))
  return api.post<ResumeAssistantResponse>('/resume/assistant', form)
}

File: backend/app/schemas/resume.py
from typing import Literal, Optional

from pydantic import BaseModel, Field


class ResumePolishRequest(BaseModel):
    section: str = Field(..., min_length=1, max_length=40)
    content: str = Field(..., min_length=1, max_length=3000)
    job_description: Optional[str] = Field(None, max_length=3000)


class ResumePolishResponse(BaseModel):
    text: str


class ResumeAnalyzeRequest(BaseModel):
    resume_text: str = Field(..., min_length=1, max_length=8000)
    job_description: str = Field(..., min_length=1, max_length=6000)


class ResumeAnalyzeResponse(BaseModel):
    analysis: str


class ResumeAssistantMessage(BaseModel):
    role: Literal["user", "assistant"]
    content: str = Field(..., min_length=1, max_length=3000)


class ResumeAssistantRequest(BaseModel):
    text: Optional[str] = Field(None, max_length=3000)
    resume_text: str = Field("", max_length=10000)
    job_description: Optional[str] = Field(None, max_length=6000)
    interview_role: Optional[str] = Field(None, max_length=80)
    history: list[ResumeAssistantMessage] = Field(default_factory=list, max_length=30)
    enable_tts: bool = True


class ResumeAssistantResponse(BaseModel):
    text: str
    user_text: str
    mode: Literal["resume", "interview", "interview_summary"]
    interview_question_index: Optional[int] = None
    interview_total: int = 10
    tts_audio_url: Optional[str] = None
